### PR TITLE
added model parameter_plot

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR plots the ElasticNet parameters for all models generated. Please let me know if you have any questions

### CODE REVIEW

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+input_file_path <- file.path(paste0("../results/","regression/",cell_type))
+# read in the data
+output_path <- file.path(paste0("../figures/","regression/",cell_type,"/"))
+# create output directory if it doesn't exist
+dir.create(output_path, recursive = TRUE, showWarnings = FALSE)
+
+files <- list.files(path = input_file_path, pattern = "*.csv", full.names = TRUE)
+
+# define empty df with column names
+model_df <- data.frame(
```
Comment (d33bs) [inline 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r:14]: Found this really neat to see, love the declarative nature here. Completely optional: is there a way to ensure the column types are received as described (casting or raising an error if not)? What would happen if the data value types were out of alignment for some reason?  [Pandera](https://github.com/unionai-oss/pandera) comes to mind within the Python landscape, but I wasn't sure if there was an R equivalent.

**Review by d33bs — CHANGES_REQUESTED** [review]
Nice work! I thought things looked good overall and mostly had suggestions towards improvements. I'm requesting changes due to the inclusion of `assert`, as I feel it's important enough to address before a merge. Please don't hesitate to reach out with any questions.

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/1.model_coefficients.ipynb [hunk 2]
```diff
@@ -46,36 +42,57 @@
...
+                ")\n",
+                "model_stats_path = pathlib.Path(\n",
+                "    f\"../../2.test_model/results/regression/{cell_type}/aggregated_with_nomic/\"\n",
                 ")"
             ]
         },
         {
             "cell_type": "code",
-            "execution_count": 5,
+            "execution_count": 15,
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/1.model_coefficients.ipynb:53]: I will re-run and re-commit

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/3.model_parameter_visualization.ipynb
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/3.model_parameter_visualization.ipynb:1]: Added!

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/3.model_parameter_visualization.ipynb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
...DWys3NveOOO0yunJycXFVV9dOf/nThwoUrVqxQFOXNN9989NFHc3JyXnvttV27dnXu3PnRRx/985//rCjKpk2bJk+erCjKiRMnXn311at+1e7duydMmKAoys6dO995552777574cKFixYtas3HBgCAzXHLKACgDXG5XGfPnn377be/973vGSMXL14cO3bsrl273G63oigjR47s2LHj66+/PmHChN69e3fq1ElRFOPy4FWWLl3avXv3/fv3p6SkKIoyduzY8vLyjRs3Ll26tGPHjgl8TAAAtF1cIQQAtC0ul2vixImNi4sWLdq3b5+RBhVFycjIuPnmmz///PPYv+TKlSvnz5/Py8sz0qBhwoQJqqoeO3bMirIBALAjAiEAoG3p3Lmzy/W/G1iqq6ufeeaZ2267LTMz0+PxeDyeS5cuaZoW+5dcvnxZUZTu3bs3HezWrZuiKP/5z38sqBoAAFvillEAQNty1SeL/uAHPzh06NDzzz8/evTojIwMh8Nx7733XveXGN9F0dDQ0HTQ+ESZpl9TAQCA5AiEAIC268qVK3/5y1/mzZu3ZMkSYyQSifj9/utumJOToyhKaWlp08FLly4pzS4bAgAgM24ZBQC0XYFAQFGUpp8Bs3379rq6umg0aiwal/sikchVG3bp0uUb3/jGnj17QqFQ4+Du3btTU1O//e1vW143AAA2wRVCAIC1jhw5cv78eeMHRVH27NljLI4bN864jhdDr169cnJytm3bNnDgwOzs7HfffffMmTOjR48+efLkwYMHhw4darwt8Be/+MXtt9/+0EMPNd129erVkyZNuv/++xcsWJCcnPz666/v3bt39erVGRkZVj1UAADsxsFX9AIAWt2IESPKy8uN4Dd79uxXXnml+TrvvffeAw88cNXgfffdV1JScuHChcaRkydPzps3r6ioKD09fcqUKWvXrj18+PCMGTPcbvfhw4e9Xu+kSZPOnz/fr1+/06dPDxs2rLq62vi7iqLs27dv1apVp0+fjkQit99++1NPPTVjxgyrHjMAADZEIAQAAAAASfEeQgAAAACQFIEQAAAAACRFIAQAAAAASREIAQAAAEBSBEIAAAAAkBSBEAAAAAAkRSAEAAAAAEkRCAEAAABAUgRCAAAAAJAUgRAAAAAAJEUgBAAAAABJ/R84kx9DjUA2tQAAAABJRU5ErkJggg==",
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/3.model_parameter_visualization.ipynb:100]: There is one outlier that is shifting the alpha color bar... I will remove the color bar

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/1.model_coefficients.py [hunk 4]
```diff
@@ -35,32 +31,53 @@
...
+    basename = pathlib.Path(model_file).name
+    basename = basename.split("__")[0] + "_variance_r2_stats.csv"
+    test_results_file = pathlib.Path(model_stats_path / basename)
+    # load the test results
+    df_test_results = pd.read_csv(test_results_file)
+    # get the r2 score
+    r2 = df_test_results.loc[(df_test_results["data_split"] == "test_data")][
+        "r2"
+    ].unique()[0]
+    r2 = np.float64(r2.strip("[]"))
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/1.model_coefficients.py:53]: Yes, I noticed this and the error is in the training script which takes a few days to run. I will be re-running this workflow again to retrain as there are other errors present too.

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/1.model_coefficients.py [hunk 5]
```diff
@@ -35,32 +31,53 @@
...
+    assert r2.dtype == np.float64
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/1.model_coefficients.py:54]: I will remove for now

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r (hunk 1 above)
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r:14]: @d33bs  I believe R raises an error something along the lines of  "cannot coerce data type" if a column is predefined as a `numeric` type and a `string` is passed.  The predefined column types help with ensuring that the `numeric` values stay numeric. There is a level deeper here in R too with `dbl` floats and `int` and such.

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/1.model_coefficients.ipynb
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/1.model_coefficients.ipynb:1]: added!

File: 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r (hunk 1 above)
Comment (d33bs) [reply 6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r:14]: Thanks for the reply on this - interesting stuff!

**Review by d33bs — APPROVED** [review]
Thank you for addressing all the comments, LGTM!

**Pull request merged**